tests/aws/services/lambda_/test_lambda.py

```diff
@@ -1867,140 +1867,163 @@ def check_logs():
-        layer_arn = primary_client.publish_layer_version(
-            LayerName=layer_name, Content={"ZipFile": dummylayer}
-        )["LayerArn"]
+        statement_id = f"stm-{short_uid()}"
+        primary_client.add_permission(
+            FunctionName=func_name,
+            StatementId=statement_id,
+            Principal=secondary_account_id,
+            Action="lambda:*",
+        )
+        yield func_arn
 
-        # Operations related to Lambda layers
+    @pytest.fixture
+    def created_layer(
+        self, primary_client, create_lambda_function, dummylayer, secondary_account_id
+    ):
+        # As of 2024-02, AWS restricts the input for adding resource-based policy to layer versions via AddLayerVersionPermission.
+        # Only 'lambda:GetLayerVersion' is accepted for Action.
+        # https://github.com/boto/botocore/blob/cf7b8449643187670620ab699596ca785e3ec889/botocore/data/lambda/2015-03-31/service-2.json#L3906-L3909
+        # This appears to have been the case at least since 2021-06.
+        # Furthermore this contradicts with what its docs on valid IAM actions for layer versions:
+        # https://docs.aws.amazon.com/lambda/latest/dg/lambda-api-permissions-ref.html#permissions-resources-layers
+
+        # Operations related to Lambda layers supported by cross account
         # - GetLayerVersion
         # - GetLayerVersionByArn
+
+        # Operations not supported by lambda layer cross account
         # - ListLayerVersions
         # - DeleteLayerVersion
         # - AddLayerVersionPermission
         # - GetLayerVersionPolicy
         # - RemoveLayerVersionPermission
 
-        # Run assertions using secondary test credentials
-
-        assert secondary_client.get_layer_version(LayerName=layer_arn, VersionNumber=1)
-
-        assert secondary_client.get_layer_version_by_arn(Arn=layer_arn + ":1")
-
-        assert secondary_client.list_layer_versions(LayerName=layer_arn)
+        layer_name = f"layer-{short_uid()}"
+        layer_arn = primary_client.publish_layer_version(
+            LayerName=layer_name, Content={"ZipFile": dummylayer}
+        )["LayerArn"]
 
-        assert secondary_client.add_layer_version_permission(
+        layer_statement_id = f"stm-{short_uid()}"
+        primary_client.add_layer_version_permission(
             LayerName=layer_arn,
-            VersionNumber=1,
+            StatementId=layer_statement_id,
+            Principal=secondary_account_id,
             Action="lambda:GetLayerVersion",
-            Principal="*",
-            StatementId="s1",
-        )
-
-        assert secondary_client.get_layer_version_policy(LayerName=layer_arn, VersionNumber=1)
-
-        assert secondary_client.remove_layer_version_permission(
-            LayerName=layer_arn, VersionNumber=1, StatementId="s1"
+            VersionNumber=1,
         )
 
-        assert secondary_client.delete_layer_version(LayerName=layer_arn, VersionNumber=1)
+        yield layer_arn
```

Comment:
```suggestion
        return layer_arn
```
Here as well, no reason to yield when there is no code after it.

Reply:
Fixed